src/hub/action_request.ts

```diff
@@ -118,6 +133,92 @@ export class ActionRequest {
   webhookId?: string
   lookerVersion: string | null = null
 
+  /** `stream` creates and manages a stream of the request data
+   *
+   * ```ts
+   * let prom = await request.stream(async (readable) => {
+   *    return myService.uploadStreaming(readable).promise()
+   * })
+   * ```
+   *
+   * Streaming generally occurs only if Looker sends the data in a streaming fashion via a push url,
+   * however it will also wrap non-streaming attachment data so that actions only need a single implementation.
+   *
+   * @returns The return value of the `callback` function. This is useful for returning
+   * a promise from the `callback` function.
+   * @param callback A function will be caled with a Node.js `Readable` object.
+   * The readable object represents the streaming data.
+   */
+  stream<T>(callback: (readable: Readable) => T): T {
+    const url = this.scheduledPlan && this.scheduledPlan.downloadUrl
```

Comment:
I have a PR that can and will change the downloadUrl location to the attachment in helltool in waiting, that would be an easy thing to add after I am back next week though - I could just edit this part and make the change if it goes in though

Reply:
Haven't we shipped it in the previous place though?

Reply:
Will that be a backwards compatible change?

Reply:
Ah I was changing the location as a response to this: https://github.com/looker/helltool/pull/37008 . I can just undo that change. It won't be backwards compatible unless I put it in both the scheduledPlan and the attachment